### DISCUSSION TITLE
Problem with codecs (I think)

### DISCUSSION THREAD

**Discussion by gabichulas** [jellyfin/jellyfin#11433] [Q&A] [answered]
Hey! I just downloaded and configured Jellyfin and some other services and I'm very happy with the result, but I have a _problem_.

Some videos (movies and tv shows) stutter _(unwatchable)_ and I can't understand why. I don't know anything about codecs and transcoding, so what I'm saying might be wrong, but yesterday I figured out that every movie/tv show file stuttering was in HEVC codec; I fixed this really easy by filtering this word and just downloading h264 files. 

I thought that it would be the solution, but here I am. It turns out that some h264 files **stutter too**, and I don't know what to do. It happens in every dispositive (browser, TV ans android phone). The files have some differences between their file information and I think the problem should be there, but as I said, I'm a complete newbie. 

![Screenshot_20240425-114202_Jellyfin.jpg](https://github.com/jellyfin/jellyfin/assets/64572010/4f3586e4-e63f-42ec-8a40-9f17a7f4d9d9)

This is an example of a h264 file that plays **perfectly**.

![Screenshot_20240425-114223_Jellyfin.jpg](https://github.com/jellyfin/jellyfin/assets/64572010/b3a6a02b-82e9-4b1b-898d-01fccfd5e08a)

And this is an example of a h264 file that stutters.

Another thing that is maybe wrong, but catches my attention, is that, for example, the file that stutters in the SS above is a B99 episode (400mb) and another movie (not the one of the screenshot, but nearly the same file info) that works fine is 16GB!!! 

Help!

**Comment by TimGels** [comment, marked as answer]
Hi there!

Your second file's video track is using the H264 (AVC) codec. While this has very broad hardware decoding support, that is only true for the 8 bit depth version of this codec. You can see that the second screenshot's video file is using 10 bit depth, which almost no consumer device has hardware decoding support for. I only know of Amazon Fire TV Max 4k (or what the name is) having support for 10 bit H264. 

This is most likely the cause for your troubled playback. If you have transcoding enabled, this would usually trigger transcoding on the Jellyfin server side. If your server is not fast enough for transcoding, then it could result in stuttery playback. But if your client does not tell the Jellyfin server to start transcoding, and it stutters, it means your client is not fast enough to decode it. I do not know what clients you are using, but I assume you are using the official clients on their default settings (no external players etc,.). And I am assuming that you have CPU transcoding enabled on the server. As a result my guess is that your server is not fast enough to be able to transcoding 10 bit H264 to the more widely supported 8 bit H264 format.

Let me know if this answer helps you and if it answers your question!

**Reply by gabichulas** [reply]
Thank you so much! I didn't know that little number could break everything, lol. You're right, my server is a very low budget laptop that I saved from the trash and I'm experimenting with it. I think my only solution is to verify manually the bit depth of every tv show I download. 

Thanks!

**Reply by TimGels** [reply]
Be aware that it is mostly 10 bit h264/avc that is having poor hardware decoding support. 10 bit h265/hevc has very wide hardware decoding support. It of-course still depends on your client and client device.
